userfiles/civet_virtual_study/civet_virtual_study.rb

```diff
@@ -0,0 +1,82 @@
+
+#
+# CBRAIN Project
+#
+# Copyright (C) 2008-2012
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# This class represents a FileCollection meant specifically
+# to represent the output of several *CIVET* runs (see CbrainTask::Civet).
+#
+class CivetVirtualStudy < CivetStudy
+
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  has_viewer :name => 'CIVET Virtual Study', :partial => :civet_virtual_study
+
+  # Sync the CivetVirtualStudy, with the CivetOutputs if deep=true
+  def sync_to_cache(deep=true)
+    super()
+    if deep
+      civet_outputs = get_civet_outputs
+      make_cache_symlinks(civet_outputs)
+    end
+  end
+
+  # Load the list of CivetOutputs from the CbrainFileList
+  def get_civet_outputs
+    self.sync_to_cache(false)
```

Comment:
Without this logic, the programmer has to have detailed knowledge of how the study operates, and I was trying to avoid that so that functions work the way they're expected to. `sync_to_cache` without `deep=true` is a very inexpensive operation, it's just a .cbcsv file.